base/deprecated.jl

```diff
@@ -2062,6 +2062,57 @@ end
 # deprecate bits to bitstring (#24263, #24281)
 @deprecate bits bitstring
 
+# deprecate speye
```

Comment:
Should `export speye` to avoid #24325 again :)

Reply:
Good catch! One day I will remember :). Also fixed. Thanks! :)